models/user.js

```diff
@@ -172,12 +178,27 @@ UserSchema.methods.loginError = function() {
         message: "You are banned from using this service due to violations of the rules.",
         code: "banned"
     };
-    if (this.deactivated === true) return {
-        message: "Your account has been deactivated. Please contact the moderators via Discord to reactivate your account.",
-        code: "deactivated"
-    };
     return null;
 }
+
+// Helper
+Date.prototype.addDays = function(days) {
+    var dat = new Date(this.valueOf());
+    dat.setDate(dat.getDate() + days);
+    return dat;
+}
+
+UserSchema.methods.markForDeletion = function() {
+    const deletionDate = Date().addDays(7);
```

Comment:
why is this defined like this if its literally used 2 lines below, and only once????

Reply:
readability